handler/handler.go

```diff
@@ -87,18 +86,101 @@ func (h *Handler) DeleteAnnotation(w http.ResponseWriter, r *http.Request) {
 	}
 	uppList = uppList[:i]
 
+	writeAnnotations(ctx, h, w, uppList, writeLog, oldHash, contentUUID)
+}
+
+// AddAnnotation adds an annotation for a specific content uuid.
+// It gets the annotations only from UPP skipping V2 annotations because they are not editorially curated.
+func (h *Handler) AddAnnotation(w http.ResponseWriter, r *http.Request) {
+	w.Header().Add("Content-Type", "application/json")
+
+	contentUUID := vestigo.Param(r, "uuid")
+
+	tID := tidutils.GetTransactionIDFromRequest(r)
+	ctx := tidutils.TransactionAwareContext(context.Background(), tID)
+
+	writeLog := log.WithField(tidutils.TransactionIDKey, tID).WithField("uuid", contentUUID)
+
+	if err := validateUUID(contentUUID); err != nil {
+		handleWriteErrors("Invalid content UUID", err, writeLog, w, http.StatusBadRequest)
+		return
+	}
+
+	annotation := annotations.Annotation{}
+	dec := json.NewDecoder(r.Body)
+	err := dec.Decode(&annotation)
+	if err != nil {
+		handleWriteErrors("Error decoding request body", err, writeLog, w, http.StatusInternalServerError)
+	}
+
+	oldHash := r.Header.Get(annotations.PreviousDocumentHashHeader)
+
+	if err := validateUUID(annotation.ConceptId); err != nil {
+		handleWriteErrors("Invalid concept UUID", err, writeLog, w, http.StatusBadRequest)
+		return
+	}
+
+	if err := validatePredicate(annotation.Predicate); err != nil {
+		handleWriteErrors(err.Error(), err, writeLog, w, http.StatusBadRequest)
+		return
+	}
+
+	uppList, err := getAnnotations(ctx, h, w, writeLog, contentUUID)
+	if err != nil {
+		handleErrors(err, writeLog, w)
+		return
+	}
+
+	conceptUUID := mapper.TransformConceptID("/" + annotation.ConceptId)
+	annotation.ConceptId = conceptUUID
+
+	uppList = canonicalizeAnnotations(h, uppList, writeLog)
+
+	for _, item := range uppList {
+		if conceptUUID == item.ConceptId {
+			if annotation.Predicate == item.Predicate {
+				writeLog.Debug("Annotation is already in list")
+				uppList = canonicalizeAnnotations(h, uppList, writeLog)
+				writeAnnotations(ctx, h, w, uppList, writeLog, oldHash, contentUUID)
+				return
+			}
+			break
+		}
+	}
+
+	uppList = append(uppList, annotation)
+	uppList = canonicalizeAnnotations(h, uppList, writeLog)
+	writeAnnotations(ctx, h, w, uppList, writeLog, oldHash, contentUUID)
+}
+
+func getAnnotations(ctx context.Context, h *Handler, w http.ResponseWriter, writeLog *log.Entry, contentUUID string) ([]annotations.Annotation, error) {
```

Comment:
`getAnnotations`, `canonicalizeAnnotations`, `writeAnnotations` should not receive `Handler` as parameter. It should be receiver. What I mean is that they should be methods of the `Handler` struct.